### DISCUSSION TITLE
Flipping the streamed image from the Front camera

### DISCUSSION THREAD

**Discussion by replicant1** [pedroSG94/RootEncoder#1561] [General]
Hi Pedro,

When I stream from the front camera I note that the video image in the preview and the video image as streamed are consistent with each other (which is desirable) but **both** images happen to be horizontally flipped from what I would prefer. For my use case, I would like to find some way to horizontally flip the **streamed** image when using the front camera. I don't mind whether the **preview** image is horizontally flipped or not. 

I've looked through the RootEncoder API and I can't see any support for doing this - but maybe I'm missing something? I also wonder about the relevance of the "filters" concept you have - could I impose a filter to do the horizontal flip - would that change the streamed image or only the preview image?

Thanks again for your advice,

Rod.

**Comment by pedroSG94** [comment]
Hello,

If you are using OpenGlView or StreamBase you can flip horizontally and vertically in preview or in stream result as you want. 
Code example:
```kotlin
    rtmpCamera1.glInterface.setIsPreviewVerticalFlip(true)
    rtmpCamera1.glInterface.setIsStreamVerticalFlip(true)
    rtmpCamera1.glInterface.setIsPreviewHorizontalFlip(true)
    rtmpCamera1.glInterface.setIsStreamHorizontalFlip(true)
```
You can do it on fly so you can change the value after switch camera if you only want do it for front camera

**Reply by replicant1** [reply]
Fantastic! Thanks Pedro!
